bftengine/src/bftengine/SimpleClientImp.cpp

```diff
@@ -213,9 +215,9 @@ int SimpleClientImp::sendRequest(uint8_t flags,
 
   ClientRequestMsg* reqMsg;
   if (isPreProcessRequired)
-    reqMsg = new preprocessor::ClientPreProcessRequestMsg(clientId_, reqSeqNum, lengthOfRequest, request);
+    reqMsg = new preprocessor::ClientPreProcessRequestMsg(clientId_, reqSeqNum, lengthOfRequest, request, msgCid);
   else
-    reqMsg = new ClientRequestMsg(clientId_, flags, reqSeqNum, lengthOfRequest, request);
```

Comment:
This looks like a reversion of a prior change made for pre-execution.

Reply:
Fixed

Reply:
This still looks wrong. It should take `flags` and not `isReadOnly`.

Reply:
Fixed